src/main/java/com/avdo/Program.java

```
@@ -14,9 +14,9 @@ public static void main(String[] args) {
         PhoneNumber p6 = null;
         PhoneNumber p7 = null;
         PhoneNumber p8 = null;
-        PhoneNumber s = MobilePhoneNumber.createNumber(62, null); // does something like this make sense?
-        createInstance(p1, 62, null);                            // or something like this...
-        PhoneNumber s1 = MobilePhoneNumber.createNumber(62, "111-111");
+        PhoneNumber s = MobilePhoneNumber.createNumber(62, null); // does something like this make sense? In sense that constructor is throwing exception, and I want try block to continue with executing in case something in it throw that exception
```

We never want to allow instances which are in invalid state. `MobilePhoneNumber.createNumber` should just propagate the exception to the caller (in this case, caller is the main method). The caller can have a try-catch block and then decide what to do. However, these values are hardcoded and it is hard to see the purpose of that, but if we imagine that some human is inputting `provider` and `number` values than we could catch the exception, write message "Provider is invalid - please enter appropriate provider number" and then retry the input.

```java
do {
  int provider = readProviderNumberFromUser();
  String phoneNumber = readPhoneNumberFromUser();
  boolean inputSuccess = false;

  try {
    PhoneNumber p = MobilePhoneNumber.createNumber(provider, phoneNumber);
    inputSuccess = true;
  } catch(SomeNumberFormatException e) {
    print("Invalid parameters for phone number, provider=%s, number=%s. Try again tralalal"
            .formatted(provider, phoneNumber);
  }

while (!inputSuccess);